clicommand/agent_start.go

```diff
@@ -658,7 +658,7 @@ var AgentStartCommand = cli.Command{
 		},
 		cli.StringFlag{
 			Name:   "signing-jwks-file",
-			Usage:  "Path to a file containing a signing key. Passing this flag enables pipeline signing for all pipelines uploaded by this agent. For hmac-sha256, the raw file content is used as the shared key",
+			Usage:  "Path to a file containing a signing key. Passing this flag enables pipeline signing for all pipelines uploaded by this agent. For hmac-sha256, the raw file content is used as the shared key. If using Docker containers to upload pipeline steps dynamically, turn on environment propagation to allow signing for the new steps generated.",
```

Comment:
Hi @PriyaSudip , I had a couple of simple queries about this addition of yours:

- Whereabouts is 'environment propagation' turn on? Is this somewhere in the Buildkite interface. If not, whereabouts exactly is this done?
- Would it be more correct to refer to 'environment propagation' as 'environment variable propagation' (and whereabouts to is the propagation happening)? Otherwise, if it's not environment variables which are being propagated, then what aspect of 'environment' is being propagated?
- Last, what exactly is being signed (a pipeline or a step of a pipeline)?

Reply:
I can take a stab at answering these:

* "Environment variable propagation" (which I agree is better than "environment propagation") refers to the `docker` CLI `-e` flag (or equivalent configuration in `docker compose` or the docker/docker compose plugins, or whatever the user is using), i.e. the user must ensure Docker/their container system is configured to set the signing-related environment variables to the same values _within_ the container that runs `buildkite-agent pipeline upload` as the agent has set _outside_ the container.
* All the command steps within the pipeline are signed when `buildkite-agent pipeline upload` is run with this configuration (https://buildkite.com/docs/agent/v3/signed-pipelines).

Reply:
Thanks very much for clarifying this @DrJosh9000 ! I'll revise this wording accordingly.